nix/pkgs/aws-ofi-nccl/cleanSource.nix

```diff
@@ -0,0 +1,76 @@
+{ lib, self }:
+let
+  inherit (lib.fileset)
+    intersection
+    difference
+    unions
+    fileFilter
+    fromSource
+    toSource
+    gitTracked
+    traceVal
+    ;
+  inherit (builtins)
+    any
+    ;
+
+  dirs = {
+    third-party = ./../../../3rd-party;
+    docs = ./../../../doc;
+    headers = ./../../../include;
+    mfour = ./../../../m4;
+    nix = ./../../../nix;
+    tus = ./../../../src;
+    tests = ./../../../tests;
+    topologies = ./../../../topology;
+  };
+
+  sourceFilter = fileFilter (
+    file:
+    any file.hasExt [
+      "c"
+      "cc"
+      "cpp"
+      "h"
+      "hpp"
+      "hh"
+      "xml"
+    ]
+  );
+
+  buildFileFilter = fileFilter (
+    file:
+    any file.hasExt [
+      "in"
+      "m4"
+      "ac"
+      "am"
+    ]
+  );
+
+  cleanRepo = traceVal (gitTracked ../../../.);
+  cleaned = x: intersection x (gitTracked ../../../.);
+  sourceFiles = cleaned (sourceFilter ../../../.);
+  buildFiles = cleaned (buildFileFilter ../../../.);
+  thirdPartyFiles = cleaned ../../../.;
```

Comment:
Would recommend adding `root = ../../..` to the `let in` block here and then using that instead of `../../../.` (which is equivalent) everywhere. At the end of the `let in` you can then do `inherit root;` in the `lib.fileset.toSource` call.